pkgs/applications/science/electronics/kicad/macos-disable-codesigning.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 1007d38..c945de2 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1057,7 +1057,7 @@ if( KICAD_BUILD_I18N )
+ endif()
+ 
+ if( APPLE )
+-    set( KICAD_OSX_CODESIGN ON
++    set( KICAD_OSX_CODESIGN OFF
```

Comment:
IIRC that's what I tried first; I'm not super familiar with CMake but my understanding is that the `FORCE` here makes it so that CLI provided values for this option are shadowed:

https://github.com/NixOS/nixpkgs/blob/f0f3b5e609be6a54fe22a0cfc035a8b64de8022a/pkgs/applications/science/electronics/kicad/macos-disable-codesigning.patch#L10-L11

Reply:
In that case, we might want to contribute a fix that removes FORCE to upstream.

Reply:
Unfortunately I don't think I have the bandwidth or the context to make an MR for this in upstream. I'm also not sure they'd be amenable to a change like this (wanting to disable explicit codesigning on macOS because you have a compiler wrapper that handles it for you isn't necessarily unique to nixpkgs but it definitely seems like a niche use case).

FWIW I think this is actually the least worrisome of the three patches (from a maintenance perspective) but if you'd prefer I can try to add `codesign` as a dep and let the KiCad build re-sign the binaries (they pass `--force` so the re-signing part should be okay but I'm not sure that `codesign`'s `sigtool` is capable of signing `.app` bundles) in lieu of the patch.